src/expr.rs

```diff
@@ -812,9 +821,9 @@ fn rewrite_if_else(context: &RewriteContext,
         try_opt!(write!(&mut result,
                         "{}else{}",
                         between_if_else_block_comment.as_ref()
```

Comment:
Hmm, that's a good counter-example. Should probably look at some more code examples (I think the heuristic for leaving code on multiple lines if it was originally gets in the way of getting a good idea of the trade-off because it means examples with short lines don't get put on a single line). I was thinking of tweaking or dumping that heuristic in any case...